dojo/api_v2/permissions.py

```diff
@@ -278,30 +279,24 @@ def has_permission(self, request, view):
         test = get_target_test_if_exists(test_id, test_title, scan_type, engagement)
 
         if test:
-            # existing engagement, nothing special to check
+            # existing test, nothing special to check
             return user_has_permission(request.user, test, Permissions.Import_Scan_Result)
         elif test_id:
-            # engagement_id doesn't exist
+            # test_id doesn't exist
             raise serializers.ValidationError("Test '%s' doesn't exist" % test_id)
 
-        if engagement and engagement_name:
-            if not auto_create_context:
-                if test_title:
-                    raise serializers.ValidationError("Test '%s' with scan_type '%s'  doesn't exist in Engagement %s" % (scan_type, test_title, engagement_name))
-                if scan_type:
-                    raise serializers.ValidationError("Test with scan_type '%s' doesn't exist in Engagement %s" % (scan_type, engagement_name))
-                raise serializers.ValidationError("No test founf in Engagement %s" % engagement_name)
-
-            if not user_has_permission(request.user, engagement, Permissions.Import_Scan_Result):
-                raise PermissionDenied('No permission to import scans into engagement %s', engagement_name)
-
-            return True
-
-        if check_import_engagement_permission(request.user, product, product_name, engagement_name, auto_create_context):
-            return True
+        if not auto_create_context:
+            # avoid NoneType errors on logging
+            if not engagement_name:
+                engagement_name = ''
 
-        if check_import_product_permission(request.user, product, product_name, product_type, product_type_name, auto_create_context):
-            return True
+            if test_title:
+                raise serializers.ValidationError("Test '%s' with scan_type '%s'  doesn't exist in Engagement %s" % (scan_type, test_title, engagement_name))
+            raise serializers.ValidationError("Test with scan_type '%s' doesn't exist in Engagement %s" % (scan_type, engagement_name))
+        else:
+            # the test doesn't exist, so we need to check if the user has requested and is allowed to use auto_create
+            if check_auto_create_permission(request.user, product, product_name, engagement, engagement_name, product_type, product_type_name):
+                return True
```

Comment:
Not sure how to reach the error message in line 301, but it's not correct. You don't need the `test_title`. It can be empty when the `scan_type` is correct. _[Couldn't put a comment on line 301]_